src/BlazorWebFormsComponents/ComponentIdGenerator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace BlazorWebFormsComponents
+{
+	/// <summary>
+	/// Represents a utility class to generate the component Id.
+	/// </summary>
+  public class ComponentIdGenerator
+  {
+		/// <summary>
+		/// Generates an Id for a given component.
+		/// </summary>
+		/// <param name="component">The <see cref="BaseWebFormsComponent"/>.</param>
+		/// <returns>The generated component Id.</returns>
+		/// <remarks>The generated Id format is {ComponentType}_{GUID} if the <see cref="BaseWebFormsComponent.ID"/> is not set.</remarks>
+		/// <example>Login_8a1112a8-bdf3-4a3c-91f6-caa9ee1ff974.</example>
+		public static string Generate(BaseWebFormsComponent component)
+		{
+			// TODO: Use ClientIDMode
+			var generatedId = component.ID;
+			if (!string.IsNullOrEmpty(generatedId))
+			{
+				return generatedId;
+			}
+
+			generatedId = component.AdditionalAttributes.ContainsKey("name")
+						? component.AdditionalAttributes["name"].ToString()
+						: GenerateId(component);
+
+			return generatedId;
+
+			string GenerateId(BaseWebFormsComponent component)
+				=> $"{component.GetType().Name}_{Guid.NewGuid().ToString()}";
```

Comment:
Web Forms didn't use a GUID for the counter, but rather used an incrementing integer.  Can we use that instead?

Reply:
I already did that, but the issue I faced is I'm unable to know the parent page so the counter increment per page not for all the pages